src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -38,13 +38,15 @@ public JavaScriptIndex(IndexDefinition definition, RavenConfiguration configurat
                 options.LimitRecursion(64)
                     .SetReferencesResolver(resolver)
                     .MaxStatements(configuration.Indexing.MaxStepsForScript)
-                    .Strict()
                     .AddObjectConverter(new JintGuidConverter())
                     .AddObjectConverter(new JintStringConverter())
                     .AddObjectConverter(new JintEnumConverter())
                     .AddObjectConverter(new JintDateTimeConverter())
                     .AddObjectConverter(new JintTimeSpanConverter())
                     .LocalTimeZone(TimeZoneInfo.Utc);
+
+                if (configuration.Patching.StrictMode)
+                    options.Strict();
```

Comment:
It's `Strict(bool strict = true)` so you could chain it as well

Reply:
Done